run_unittests.py

```diff
@@ -9864,6 +9868,27 @@ def ran_in(s):
         out = self._subprojects_cmd(['foreach', '--types', 'git'] + dummy_cmd)
         self.assertEqual(ran_in(out), ['subprojects/sub_git'])
 
+    def test_purge(self):
+        self._create_project(self.subprojects_dir / 'sub_file')
+        self._wrap_create_file('sub_file')
```

Comment:
Would be nice to also create a wrap git here, that would have catched the error I found.